### PR TITLE
net: implement load_balancing_algorithm::fixed

### PR DESCRIPTION
**Pull request by cyx1231st**
Add a new load_balancing_algorithm::fixed to force server_socket to accept connected_socket on a fixed CPU.

### CODE REVIEW

File: src/net/posix-stack.cc [hunk 1]
```diff
@@ -427,8 +427,17 @@ posix_server_socket_impl::accept() {
     return _lfd.accept().then([this] (std::tuple<pollable_fd, socket_address> fd_sa) {
         auto& fd = std::get<0>(fd_sa);
         auto& sa = std::get<1>(fd_sa);
-        auto cth = _lba == server_socket::load_balancing_algorithm::connection_distribution ?
-                _conntrack.get_handle() : _conntrack.get_handle(ntoh(sa.as_posix_sockaddr_in().sin_port) % smp::count);
+        auto cth = [this, &sa] {
+            switch(_lba) {
```
Comment (rzarzynski) [inline src/net/posix-stack.cc:431]: :+1: for improving the code readability.

**Review by rzarzynski — COMMENTED** [review]
I would advice to split the PR into two commits:
   * the clean-up,
   * introduction of `load_balancing_algorithm::fixed`

and send the former one without waiting on the latter.

**Comment by cyx1231st** [comment]
Updated related documentation. I didn't split the commit because I think it's acceptable to use conditional ternary operator when there are only two cases, except that the line is too long and could be wrapped.

**Comment by tchaikov** [comment]
closing, as it's been upstreamed.

File: src/net/posix-stack.cc (hunk 1 above)
Comment (tchaikov) [reply src/net/posix-stack.cc:431]: i'd suggest add a space after `switch`.

**Pull request closed without merging**